src/analysis/complex/polynomial.lean

```diff
@@ -1,99 +1,39 @@
-  from (div_lt_one hf0').2 $ (lt_div_iff' hg0').1 $
-  calc δ ^ n ≤ δ : pow_le_of_le_one (le_of_lt hδ0) hδ1 hn0
-         ... ≤ ((f.eval z₀).abs / (g.eval z₀).abs) / 2 : min_le_right _ _
-         ... < _ : half_lt_self (div_pos hf0' hg0'),
-have hF₂ : (F.eval z').abs = (f.eval z₀).abs - (g.eval z₀).abs * δ ^ n,
-  from calc (F.eval z').abs = (f.eval z₀ - f.eval z₀ * (g.eval z₀).abs
-    * δ ^ n / (f.eval z₀).abs).abs : congr_arg abs hF₁
-  ... = abs (f.eval z₀) * complex.abs (1 - (g.eval z₀).abs * δ ^ n /
-      (f.eval z₀).abs : ℝ) : by rw [←map_mul];
-        exact congr_arg complex.abs
-          (by simp only [mul_add, mul_assoc, div_eq_mul_inv, sub_eq_add_neg, of_real_add, mul_one,
-                         of_real_one, of_real_neg, of_real_mul, of_real_pow, of_real_inv, mul_neg])
-  ... = _ : by rw [complex.abs_of_nonneg (sub_nonneg.2 (le_of_lt hδs)),
-      mul_sub, mul_div_cancel' _ (ne.symm (ne_of_lt hf0')), mul_one],
-have hef0 : abs (eval z₀ g) * (eval z₀ f).abs ≠ 0,
-  from mul_ne_zero (abs.ne_zero hg0) (abs.ne_zero hf0),
-have hz'z₀ : abs (z' - z₀) = δ,
-  by simp only [z', mul_assoc, mul_left_comm _ (_ ^ n), mul_comm _ (_ ^ n), mul_comm (eval _ f).abs,
-                _root_.mul_div_cancel _ hef0, of_real_mul, neg_mul, neg_div, map_pow, abs_of_real,
-                add_sub_cancel, abs_cpow_inv_nat, absolute_value.map_neg, map_div₀, map_mul,
-                abs_abs, complex.abs_of_nonneg hδ0.le, real.pow_nat_rpow_nat_inv hδ0.le hn0],
-have hF₃ : (f.eval z' - F.eval z').abs < (g.eval z₀).abs * δ ^ n,
-  from calc (f.eval z' - F.eval z').abs
-      = (g.eval z' - g.eval z₀).abs * (z' - z₀).abs ^ n :
-        by rw [← eq_sub_iff_add_eq.1 hg, ←map_pow abs, ←map_mul, sub_mul];
-           simp only [eval_pow, eval_add, eval_mul, eval_C, eval_X, eval_neg, sub_eq_add_neg,
-                      add_assoc, neg_add_rev, add_neg_cancel_comm_assoc]
-  ... = (g.eval z' - g.eval z₀).abs * δ ^ n : by rw hz'z₀
-  ... < _ : (mul_lt_mul_right (pow_pos hδ0 _)).2 (hδ _ hz'z₀),
-lt_irrefl (f.eval z₀).abs $
-  calc (f.eval z₀).abs ≤ (f.eval z').abs : hz₀ _
-    ... = (F.eval z' + (f.eval z' - F.eval z')).abs : by simp
-    ... ≤ (F.eval z').abs + (f.eval z' - F.eval z').abs : abs.add_le _ _
-    ... < (f.eval z₀).abs - (g.eval z₀).abs * δ ^ n + (g.eval z₀).abs * δ ^ n :
-      add_lt_add_of_le_of_lt (by rw hF₂) hF₃
-    ... = (f.eval z₀).abs : sub_add_cancel _ _
+begin
+  contrapose! hf,
+  obtain ⟨c, hc⟩ := (f.differentiable.inv hf).exists_const_forall_eq_of_bounded _,
+  { convert degree_C_le,
+    refine polynomial.funext (λ z, _),
+    exacts [c⁻¹, by rw [eval_C, ← hc z, inv_inv]] },
+  { obtain ⟨z₀, h₀⟩ := f.exists_forall_norm_le,
+    simp_rw [metric.bounded_iff_subset_ball (0 : ℂ), set.range_subset_iff,
+             metric.mem_closed_ball, dist_eq, sub_zero, ← norm_eq_abs, norm_inv],
+    have := _, refine ⟨_, λ y, (inv_le_inv _ this).2 $ h₀ y⟩,
+    exacts [this.trans_le (h₀ y), norm_pos_iff.2 (hf z₀)] },
```

Comment:
Thank you! Optional minor golf:
```suggestion
  { obtain rfl : f = C c⁻¹ := polynomial.funext (λ z, by rw [eval_C, ← hc z, inv_inv]),
    exact degree_C_le },
  { obtain ⟨z₀, h₀⟩ := f.exists_forall_norm_le,
    simp only [bounded_iff_forall_norm_le, set.forall_range_iff, norm_inv],
    exact ⟨∥eval z₀ f∥⁻¹, λ z, inv_le_inv_of_le (norm_pos_iff.2 $ hf _) (h₀ _)⟩ },
```
bors d+

Reply:
Thanks!